src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -275,19 +290,24 @@ else if (msg.what == KILL_WORKER) {
 
                     ///////////////////////////
 
-                    if (queueDepth >= mConfig.getBulkUploadLimit()) {
+                    if ((returnCode >= mConfig.getBulkUploadLimit() || returnCode == MPDbAdapter.DBOutOfMemoryError) &&
+                            SystemClock.elapsedRealtime() >= mRetryAfter) {
```

Comment:
As we discussed, these test cases are already covered in HttpTest (503 test and short of memory test).